### PR TITLE
Added Azure.ContainerApp.Name

### LINKED ISSUE #2094: Container Apps should meet naming requirements
### Existing rule

_No response_

### Suggested rule

Container Apps should meet naming requirements.

Helps with identifying issues in the inner development cycle loop and increases deployment success.

### Pillar

None

### Additional context

- [Naming rules and restrictions for container app resource](https://learn.microsoft.com/azure/azure-resource-manager/management/resource-name-rules#microsoftapp)
- [Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.app/containerapps)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2094 

Added Azure.ContainerApp.Name rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: src/PSRule.Rules.Azure/rules/Azure.ContainerApp.Rule.yaml [hunk 1]
```diff
@@ -27,6 +27,29 @@ spec:
...
+  type:
+    - Microsoft.App/containerApps
+  condition:
+    allOf:
+      - name: '.'
+        greaterOrEquals: 2
+      - name: '.'
+        lessOrEquals: 32
+      - name: '.'
+        match: '^([a-z][0-9a-z-]{1,30}[0-9]|[a-z][0-9])$'
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.ContainerApp.Rule.yaml:51]: Maybe try this expression instead.

```suggestion
        match: '^([a-z][0-9a-z-]{0,30}[0-9a-z])$'
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Nice work @BenjaminEngeset. Just a few points to tweak this further. Let me know if you have any questions.

**Comment by BenjaminEngeset** [comment]
@BernieWhite looks like it didn't work, but looks like you are aware of it.

File: tests/PSRule.Rules.Azure.Tests/Azure.ContainerApp.Tests.ps1 [hunk 2]
```diff
@@ -71,3 +71,53 @@ Describe 'Azure.ContainerApp' -Tag 'ContainerApp' {
...
+        $invokeParams = @{
+            Baseline = 'Azure.All'
+            Module = 'PSRule.Rules.Azure'
+            WarningAction = 'Ignore'
+            ErrorAction = 'Stop'
+        }
+
+        $testObject = [PSCustomObject]@{
+            Name = ''
+            ResourceType = 'Microsoft.Resources/deployments'
```
Comment (BernieWhite) [inline tests/PSRule.Rules.Azure.Tests/Azure.ContainerApp.Tests.ps1:86]: Needs to be the same type as the rule.

```suggestion
            ResourceType = 'Microsoft.App/containerApps'
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
I added an issue for future support for `caseSensitive` of `match`.

The rule test is failed for another reason, see below.

**Comment by BenjaminEngeset** [comment]
> I added an issue for future support for `caseSensitive` of `match`.
> 
> The rule test is failed for another reason, see below.

Great! Fixed btw.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset looks good to me. Should be good to merge.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
See suggestions below.

**Comment by BernieWhite** [comment]
Adding that in.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**